storage/interceptors/smaap_service_instance_interceptor.go

```diff
@@ -0,0 +1,499 @@
+	var key *osbc.OperationKey
+	if len(operation.ExternalID) != 0 {
+		opKey := osbc.OperationKey(operation.ExternalID)
+		key = &opKey
+	}
+
+	pollingRequest := &osbc.LastOperationRequest{
+		InstanceID:   instance.ID,
+		ServiceID:    &serviceCatalogID,
+		PlanID:       &planCatalogID,
+		OperationKey: key,
+		//TODO no OI for SM platform yet
+		OriginatingIdentity: nil,
+	}
+
+	ticker := time.NewTicker(i.pollingInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			log.C(ctx).Errorf("Terminating poll last operation for instance with id %s and name %s due to context done event", instance.ID, instance.Name)
+			//operation should be kept in progress in this case
+			return nil
+		case <-ticker.C:
+			log.C(ctx).Infof("Sending poll last operation request %+v for instance with id %s and name %s", pollingRequest, instance.ID, instance.Name)
+			pollingResponse, err := osbClient.PollLastOperation(pollingRequest)
+			if err != nil {
+				if osbc.IsGoneError(err) && operation.Type == types.DELETE {
+					log.C(ctx).Infof("Successfully finished polling operation for instance with id %s and name %s", instance.ID, instance.Name)
+
+					operation.Reschedule = false
+					if _, err := i.repository.Update(ctx, operation, query.LabelChanges{}); err != nil {
+						return fmt.Errorf("failed to update operation with id %s to mark that next execution should be a reschedule", operation.ID)
+					}
+					return nil
+				}
+
+				return &util.HTTPError{
+					ErrorType: "BrokerError",
+					Description: fmt.Sprintf("Failed poll last operation request %+v for instance with id %s and name %s: %s",
+						pollingRequest, instance.ID, instance.Name, err),
+					StatusCode: http.StatusBadGateway,
+				}
+			}
+			switch pollingResponse.State {
+			case osbc.StateInProgress:
+				log.C(ctx).Infof("Polling of instance still in progress. Rescheduling polling last operation request %+v to for provisioning of instance with id %s and name %s...", pollingRequest, instance.ID, instance.Name)
+
+			case osbc.StateSucceeded:
+				log.C(ctx).Infof("Successfully finished polling operation for instance with id %s and name %s", instance.ID, instance.Name)
+
+				operation.Reschedule = false
+				if _, err := i.repository.Update(ctx, operation, query.LabelChanges{}); err != nil {
+					return fmt.Errorf("failed to update operation with id %s to mark that next execution should be a reschedule", operation.ID)
+				}
+
+				return nil
+			case osbc.StateFailed:
+				log.C(ctx).Infof("Failed polling operation for instance with id %s and name %s", instance.ID, instance.Name)
+				operation.Reschedule = false
+				if enableOrphanMitigation {
+					operation.DeletionScheduled = time.Now()
+				}
+				if _, err := i.repository.Update(ctx, operation, query.LabelChanges{}); err != nil {
+					return fmt.Errorf("failed to update operation with id %s after failed of last operation for instance with id %s", operation.ID, instance.ID)
+				}
+
```

Comment:
update `usable` field from error response `instance_usable` in order to forbid binding requests

Reply:
there is no OSB spec release with this yet (2.15 does not include it)